src/hdmf/build/objectmapper.py

```diff
@@ -564,7 +565,11 @@ def get_attr_value(self, **kwargs):
                 msg = ("%s '%s' does not have attribute '%s' for mapping to spec: %s"
                        % (container.__class__.__name__, container.name, attr_name, spec))
                 raise ContainerConfigurationError(msg)
-            if attr_val is not None:
+            if isinstance(attr_val, TermSetWrapper):
+                # This is when the wrapped item is an attribute
+                # Refer to h5tools.py for wrapped datasets
+                attr_val = attr_val.value
+            elif attr_val is not None:
```

Comment:
```suggestion
            if attr_val is not None:
```
Why `elif` here? I think it should be `if` so that the attribute value is processed normally after unwrapping

Reply:
It's elif because otherwise the value, in the case of data for Foo, will take a list and convert it to a string.